compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -42,8 +42,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let res = self.pointer_from_exposed_address_cast(&src, cast_ty)?;
                 self.write_immediate(res, dest)?;
             }
-
-            Misc => {
+            IntToInt | FloatToInt | FloatToFloat | IntToFloat | FnPtrToPtr | PtrToPtr => {
                 let src = self.read_immediate(src)?;
                 let res = self.misc_cast(&src, cast_ty)?;
```

Comment:
Please add a FIXME here indicating that we shouldn't have a single `misc_cast` for all of them, but handle them separately.

Reply:
I can do that once this get's merged. 👍

Reply:
Please add the FIXME now, even if you plan to fix it immediately in a follow-up. :)

Reply:
Ah good, I misunderstood your comment then. :)